packages/asset/test/RoyaltyDistribution.test.ts

```diff
@@ -0,0 +1,1176 @@
+import { ethers } from "hardhat";
+import { expect } from "chai";
+import { splitterAbi } from "./Splitter.abi";
+import { BigNumber } from "ethers";
+import {
+  generateAssetId,
+  royaltyDistribution,
+} from "./fixtures/royaltiesFixture";
+
+describe("Asset and catalyst Royalties", () => {
+  describe("Royalty distribution through splitter", () => {
+    it("should split ERC20 using EIP2981", async function () {
+      const {
+        Asset,
+        ERC20,
+        mockMarketplace,
+        ERC20AsBuyer,
+        deployer,
+        seller,
+        buyer,
+        commonRoyaltyReceiver,
+        creator,
+        AssetAsSeller,
+        manager,
+      } = await royaltyDistribution();
+      // generate token id to be minted with creator already set
+      const id = generateAssetId(creator, 1);
+      // directly minting to the seller for marketplace transaction
+      await Asset.connect(await ethers.getSigner(deployer)).mint(
```

Comment:
deployer should not have minter permissions on Asset & minting should be carried out by MINTER_ROLE for clarity

Reply:
Added a minter in the test cases.